File: kasa_android/app/src/main/java/com/example/kasa/util/CurrencyFormatter.kt
package com.example.kasa.util

import java.text.NumberFormat
import java.util.Locale

object CurrencyFormatter {
    fun format(amount: Double, currency: String = "USD"): String {
        return when (currency.uppercase()) {
            "USD" -> "$%.2f".format(Locale.US, amount)
            "EUR" -> "%.2f €".format(Locale.FRANCE, amount)
            "CDF" -> "%,.0f FC".format(Locale.FRENCH, amount)
            "XOF", "FCFA" -> "%,.0f FCFA".format(Locale.FRENCH, amount)
            else -> "$%.2f %s".format(Locale.US, amount, currency)
        }
    }

    fun formatCompact(amount: Double, currency: String = "USD"): String {
        return when {
            amount >= 1_000_000 -> "%.1fM %s".format(Locale.US, amount / 1_000_000, currency)
            amount >= 1_000 -> "%.1fK %s".format(Locale.US, amount / 1_000, currency)
            else -> format(amount, currency)
        }
    }
}

File: kasa/lib/utils/currency_formatter.dart
import 'package:intl/intl.dart';

class CurrencyFormatter {
  static final Map<String, NumberFormat> _formats = {
    'USD': NumberFormat.currency(symbol: '\$', decimalDigits: 2),
    'CDF': NumberFormat.currency(symbol: 'FC', decimalDigits: 0),
    'EUR': NumberFormat.currency(symbol: '€', decimalDigits: 2),
    'GBP': NumberFormat.currency(symbol: '£', decimalDigits: 2),
  };

  static String format(double amount, String currency) {
    final format = _formats[currency];
    if (format != null) return format.format(amount);
    return '$amount $currency';
  }

  static String formatWithConversion(double amount, String fromCurrency, String toCurrency, double exchangeRate) {
    if (fromCurrency == toCurrency) return format(amount, fromCurrency);
    
    final converted = fromCurrency == 'USD' 
        ? amount * exchangeRate 
        : amount / exchangeRate;
    
    return '${format(amount, fromCurrency)} ≈ ${format(converted, toCurrency)}';
  }

  static double convert(double amount, String fromCurrency, String toCurrency, double exchangeRate) {
    if (fromCurrency == toCurrency) return amount;
    return fromCurrency == 'USD' 
        ? amount * exchangeRate 
        : amount / exchangeRate;
  }

  static String formatCompact(double amount) {
    if (amount >= 1000000) return '${(amount / 1000000).toStringAsFixed(1)}M';
    if (amount >= 1000) return '${(amount / 1000).toStringAsFixed(1)}K';
    return amount.toStringAsFixed(0);
  }
}
